modules/cms/classes/ComponentBase.php

```diff
@@ -326,4 +337,12 @@ public function __toString()
     {
         return $this->alias;
     }
+
+    /**
+     * Returns the component's cache key
+     */
+    public function getCacheKey()
+    {
+        return 'component::' . $this->dirName;
```

Comment:
This should also have the name and alias of the component in here to be as unique as possible